stdlib/REPL/src/LineEdit.jl

```diff
@@ -2813,8 +2813,9 @@ function prompt!(term::TextTerminal, prompt::ModalInterface, s::MIState = init_s
             # try/catch block
             try
                 status = fcn(s, kdata)
-            catch e
-                @error "Error in the keymap" exception=e,catch_backtrace()
+            catch ex
+                println() # because we're on the prompt line
+                @error "Error in REPL keymap\n$(sprint(showerror, ex, catch_backtrace()))"
```

Comment:
this is also not the correct way to do structured logging

Reply:
I believe that usually the previous way prints the stacktrace nicely. Any idea why it was failing to here?

Reply:
Actually, no it doesn't.
```
julia> try
       error()
       catch ex
       @error "Error in the keymap" exception=ex,catch_backtrace()
       end
┌ Error: Error in the keymap
│   exception = (ErrorException(""), Union{Ptr{Nothing}, Base.InterpreterIP}[Ptr{Nothing} @0x000000013196c30f, Ptr{Nothing} @0x000000013196c33b, Ptr{Nothing} @0x00000001058a57bb, Ptr{Nothing} @0x00000001058c1abb, Ptr{Nothing} @0x00000001058c01ef, Ptr{Nothing} @0x00000001058c048b, Ptr{Nothing} @0x00000001058c0837, Base.InterpreterIP in top-level CodeInfo for Main at statement 1, Ptr{Nothing} @0x00000001058d8ac7, Ptr{Nothing} @0x00000001058d8847, Ptr{Nothing} @0x00000001058d975f, Ptr{Nothing} @0x000000013193fa97, Ptr{Nothing} @0x00000001319d450b, Ptr{Nothing} @0x000000013193ce77, Ptr{Nothing} @0x00000001318aecdb, Ptr{Nothing} @0x00000001058a57bb, Ptr{Nothing} @0x00000001318adb4f, Ptr{Nothing} @0x00000001318ae38f, Ptr{Nothing} @0x000000013193d387, Ptr{Nothing} @0x00000001058a57bb, Ptr{Nothing} @0x0000000131970bc3, Ptr{Nothing} @0x00000001318e0f6f, Ptr{Nothing} @0x00000001058a57bb, Ptr{Nothing} @0x00000001058b4807, Ptr{Nothing} @0x000000013942e00b, Ptr{Nothing} @0x000000013921c163, Ptr{Nothing} @0x000000013942ec6b, Ptr{Nothing} @0x00000001058a57bb, Ptr{Nothing} @0x00000001059067bf, Ptr{Nothing} @0x00000001059066d3])
└ @ Main REPL[4]:4
```